File: src/main/scala/com/barley/respository/EmailRepository.scala
package com.barley.respository

import java.lang.Long

import com.barley.model.Email
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Component

/**
  * Created by vikram.gulia on 11/25/15.
  */
@Component
trait EmailRepository extends CrudRepository[Email, Long]

File: src/main/java/com/barley/repository/EmailRepository.java
package com.barley.repository;

import com.barley.model.Email;
import org.springframework.context.annotation.Bean;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Component;

@Component
public interface EmailRepository extends PagingAndSortingRepository<Email, Long> {
}
